src/cpp/src/cache_manager.hpp

```diff
@@ -40,15 +64,79 @@ class CacheManager {
         } else {
             auto remote_context = m_core.get_default_context(device_name);
             for (size_t decoder_layer_id = 0; decoder_layer_id < m_device_config.get_num_layers(); ++decoder_layer_id) {
-                ov::Tensor key_cache = remote_context.create_tensor(device_config.get_cache_precision(),
-                                                                    device_config.get_key_cache_shape());
-                ov::Tensor value_cache = remote_context.create_tensor(device_config.get_cache_precision(),
-                                                                      device_config.get_value_cache_shape());
+                ov::Tensor key_cache = remote_context.create_tensor(m_device_config.get_cache_precision(),
+                                                                    key_cache_shape);
+                ov::Tensor value_cache = remote_context.create_tensor(m_device_config.get_cache_precision(),
+                                                                      value_cache_shape);
 
                 m_key_cache.emplace_back(key_cache);
                 m_value_cache.emplace_back(value_cache);
             }
         }
+        update_request_tensor();
+    }
+
+    void update_request_tensor() {
+        for (size_t decoder_layer_id = 0; decoder_layer_id < m_device_config.get_num_layers(); ++decoder_layer_id) {
+            m_request.set_tensor(std::string("key_cache.") + std::to_string(decoder_layer_id), m_key_cache[decoder_layer_id]);
+            m_request.set_tensor(std::string("value_cache.") + std::to_string(decoder_layer_id), m_value_cache[decoder_layer_id]);
+        }
+    }
+
+    void increase_cache(size_t num_kv_blocks) {
+        OPENVINO_ASSERT(num_kv_blocks > m_num_allocated_kv_blocks);
+        ov::Shape new_value_cache_shape = set_first_dim_and_make_static(m_device_config.get_value_cache_shape(), num_kv_blocks);
+        ov::Shape new_key_cache_shape = set_first_dim_and_make_static(m_device_config.get_key_cache_shape(), num_kv_blocks);
+
+        const std::string device_name = m_device_config.get_device();
+        ov::Coordinate start_key{0,0,0,0};
+        ov::Coordinate start_value{0,0,0,0};
+
+        if (device_name.find("GPU") == std::string::npos) {
+            for (size_t decoder_layer_id = 0; decoder_layer_id < m_device_config.get_num_layers(); ++decoder_layer_id) {
+                ov::Coordinate end_key(m_key_cache[decoder_layer_id].get_shape());
+                ov::Coordinate end_value(m_value_cache[decoder_layer_id].get_shape());
+
+                ov::Tensor key_cache(m_device_config.get_cache_precision(), new_value_cache_shape);
+                ov::Tensor value_cache(m_device_config.get_cache_precision(), new_key_cache_shape);
+                
+                // force allocation
+                std::memset(key_cache.data(), 0, key_cache.get_byte_size());
```

Comment:
Can you educate me on that:
1. Is there some kind of copy-on-write memory mapping for CPU requiring reallocation? What's the origin of that memory?
2. `copy_to()` also writes to that memory. Why wasn't `copy_to()` enough to trigger allocation?

Reply:
Luo Cheng has found the root cause of the accuracy issue. Some optimizations like AMX require a minimal shape and perform multiplying by zero on the excess data. Uninitialized tensor data contain `nan`'s, so `nan` * 0 returns non-zero invalid data. 
So we can't remove `memset()` here. But we can optimize `increase_cache()` method and do `memset()` only on the added data, which is not initialized by `copy_to()`.

Reply:
Add this to the code comment

Reply:
Done.